File: packages/sdk-flutter/lib/src/models/device_config.dart
/// Device-specific configuration from the server
///
/// Contains debug mode status and computed tracking enabled flag
class NivoStackDeviceConfig {
  /// Device ID from server (internal use)
  final String? deviceId;

  /// Short device code for support identification (e.g., "A7B3-X9K2")
  final String? deviceCode;

  /// Whether debug mode is enabled for this device
  final bool debugModeEnabled;

  /// When debug mode expires (null = no expiry)
  final DateTime? debugModeExpiresAt;

  /// Whether tracking is enabled for this device
  /// Computed based on tracking mode and debug status
  final bool trackingEnabled;

  const NivoStackDeviceConfig({
    this.deviceId,
    this.deviceCode,
    this.debugModeEnabled = false,
    this.debugModeExpiresAt,
    this.trackingEnabled = true,
  });

  /// Create from JSON response
  factory NivoStackDeviceConfig.fromJson(Map<String, dynamic> json) {
    DateTime? expiresAt;
    if (json['debugModeExpiresAt'] != null) {
      expiresAt = DateTime.tryParse(json['debugModeExpiresAt']);
    }

    return NivoStackDeviceConfig(
      deviceId: json['deviceId'] as String?,
      deviceCode: json['deviceCode'] as String?,
      debugModeEnabled: json['debugModeEnabled'] as bool? ?? false,
      debugModeExpiresAt: expiresAt,
      trackingEnabled: json['trackingEnabled'] as bool? ?? true,
    );
  }

  /// Create default config (tracking enabled)
  factory NivoStackDeviceConfig.defaults() {
    return const NivoStackDeviceConfig(
      debugModeEnabled: false,
      trackingEnabled: true,
    );
  }

  /// Convert to JSON
  Map<String, dynamic> toJson() {
    return {
      'deviceId': deviceId,
      'deviceCode': deviceCode,
      'debugModeEnabled': debugModeEnabled,
      'debugModeExpiresAt': debugModeExpiresAt?.toIso8601String(),
      'trackingEnabled': trackingEnabled,
    };
  }

  @override
  String toString() {
    return 'NivoStackDeviceConfig(code: $deviceCode, debug: $debugModeEnabled, tracking: $trackingEnabled)';
  }
}

File: packages/sdk-ios/Sources/NivoStack/Models/DeviceConfig.swift
import Foundation

/// Device-specific configuration from the server
///
/// Contains debug mode status and computed tracking enabled flag
public struct NivoStackDeviceConfig: Codable {
    /// Device ID from server (internal use)
    public let deviceId: String?
    
    /// Short device code for support identification (e.g., "A7B3-X9K2")
    public let deviceCode: String?
    
    /// Whether debug mode is enabled for this device
    public let debugModeEnabled: Bool
    
    /// When debug mode expires (nil = no expiry)
    public let debugModeExpiresAt: Date?
    
    /// Whether tracking is enabled for this device
    /// Computed based on tracking mode and debug status
    public let trackingEnabled: Bool
    
    public init(
        deviceId: String? = nil,
        deviceCode: String? = nil,
        debugModeEnabled: Bool = false,
        debugModeExpiresAt: Date? = nil,
        trackingEnabled: Bool = true
    ) {
        self.deviceId = deviceId
        self.deviceCode = deviceCode
        self.debugModeEnabled = debugModeEnabled
        self.debugModeExpiresAt = debugModeExpiresAt
        self.trackingEnabled = trackingEnabled
    }
    
    /// Create from JSON response
    public init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        deviceId = try container.decodeIfPresent(String.self, forKey: .deviceId)
        deviceCode = try container.decodeIfPresent(String.self, forKey: .deviceCode)
        debugModeEnabled = try container.decodeIfPresent(Bool.self, forKey: .debugModeEnabled) ?? false
        
        if let expiresAtString = try container.decodeIfPresent(String.self, forKey: .debugModeExpiresAt) {
            let formatter = ISO8601DateFormatter()
            debugModeExpiresAt = formatter.date(from: expiresAtString)
        } else {
            debugModeExpiresAt = nil
        }
        
        trackingEnabled = try container.decodeIfPresent(Bool.self, forKey: .trackingEnabled) ?? true
    }
    
    /// Create default config (tracking enabled)
    public static func defaults() -> NivoStackDeviceConfig {
        return NivoStackDeviceConfig(
            debugModeEnabled: false,
            trackingEnabled: true
        )
    }
    
    enum CodingKeys: String, CodingKey {
        case deviceId
        case deviceCode
        case debugModeEnabled
        case debugModeExpiresAt
        case trackingEnabled
    }
}

